src/content/login/index.tsx

```diff
@@ -1,70 +1,242 @@
+                                <Button
+                                    component={Link}
+                                    to="/reset-password"
+                                    variant="outlined"
+                                    type="button"
+                                    fullWidth
+                                >
+                                    Forgot password?
+                                </Button>
+                                <Button
+                                    component={Link}
+                                    to="/signup"
+                                    variant="outlined"
+                                    type="button"
+                                    fullWidth
+                                >
+                                    Looking for signup?
+                                </Button>
+                            </ButtonGroup>
+                        </Grid>
+                    </Grid>
                 </Card>
-            </Container>
-        </OverviewWrapper>
+            </Grid>
+        </Grid>
+    );
+};
+
+/**
+ * Component for the login form.
+ */
+const LoginForm = () => {
+    const [passwordVisible, setPasswordVisible] = useState(false);
+
+    const [rememberMe, setRememberMe] = useState(false);
+
+    const [usernameIR, passwordIR] = [
+        useRef<HTMLInputElement>(null),
+        useRef<HTMLInputElement>(null),
+    ];
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [usernameError, setUsernameError] = useState(false);
+    const [passwordError, setPasswordError] = useState(false);
+
+    const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        setUsername(usernameIR.current?.value || "");
+        setPassword(passwordIR.current?.value || "");
+    };
+
+    // Validation methods.
+    const validateUsername = () => {
+        console.log(username);
+        setUsernameError(username.length < 3);
+    };
+    const validatePassword = () => setPasswordError(password.length < 3);
+
+    // Prevents useEffect from running login code on init.
+    const [ready, setReady] = useState(false);
+    useEffect(() => setReady(true));
+
+    // Processes new login data from the form.
+    useEffect(() => {
+        if (!ready) return;
+
+        // Update the error state of the inputs.
+        validateUsername();
+        validatePassword();
+
+        // Don't continue to login logic if there are errors.
+        if (usernameError || passwordError) return;
+
+        /* TODO: Actual login code. */
+    }, [username, password]);
+
+    return (
+        <Grid
+            container
+            spacing={1}
+            alignItems="center"
+            justifyContent="center"
+            textAlign="center"
+            component="form"
+            onSubmit={handleSubmit}
+        >
+            <Grid item xs={12}>
+                <TextField
```

Comment:
Pass value and onChange handler here

Reply:
I love the idea of using `react-hook-form`, seems like it could be a good way to simplify the form creation process.